modules/ROOT/pages/batch-phases.adoc

```diff
@@ -0,0 +1,135 @@
+
+Note that changes to records that take place within a Batch Aggregator component _are not propagated_ outside of Batch Aggregator to the containing Batch Step component.
+
+image::mruntime-batch-step-process.png[Record Stepping Queue for Batch Steps]
+
+As processing takes place, each record keeps track of every Batch Step component it completes.
+
+[[batch_aggregator_processing]]
+By default, a batch job instance does not wait for all its queued records to finish processing in one Batch Step component before making them available for processing by the next. Configuring a Batch Aggregator component changes this behavior:
+
+* Behavior when the Batch Aggregator component sets a fixed size:
++
+In this scenario, the Batch Aggregator pulls records for a given batch job instance from the stepping queue into arrays a fixed size. If fewer records are available than the fixed size setting, Batch Aggregator pulls those records into the array without waiting for more. After processing records in an array, Batch Aggregator sends the records back to the stepping queue _without modifications_ caused by any processors within Batch Aggregator.
++
+A processor within Batch Aggregator must be able to receive the array of records as input.
++
+image:mruntime-batch-job-process-aggregator-fixed.png[Batch job process with Aggregator with fixed size]
++
+[calloutlist]
+. Each Batch Step component within a Batch Job component receives one or more record blocks and starts processing them in parallel.
+. After processing a record, the Batch Step component sends the record to the Batch Aggregator component for further processing.
+. The Batch Aggregator component places records into one or more arrays according to its `size` configuration. If fewer records are available from the queue than the configured size, the component pulls them into the component for processing. This scenario can occur, for example, when the size setting does not divide evenly into the number of records. The last set of records is smaller than the other sets.
+. After processing records in a given array, the Batch Aggregator component sends all the records back to the stepping queue.
+
+* Behavior when the Batch Aggregator component is configured to stream records:
++
+In this scenario, the Batch Job component sends the processed records to its Batch Aggregator component. The Batch Aggregator component loads an array with  records from the stepping queue until all records from the current Batch Step for a given batch job instance are aggregated.
```

Comment:
TODO: style guidelines only on change

Reply:
DONE: changed to 

> When streaming, the Batch Aggregator component processes records as soon as it receives them instead of putting them into record blocks. No record is exposed to the _next_ Batch Step component in the flow until all the records in the Aggregator processes all records in the _current_ Batch Step.